login-utils/last.c

```diff
@@ -521,27 +523,41 @@ static int list(const struct last_control *ctl, struct utmpx *p, time_t logout_t
 	if (r < 0)
 		mem2strcpy(domain, p->ut_host, sizeof(p->ut_host), sizeof(domain));
 
+	/*
+	 * set last displayed character to an asterisk when
+	 * user/domain/ip fields are to be truncated in non-fullnames mode
```

Comment:
Why need do the indication in both cases? That would make the code simpler, too.

Reply:
Or do you mean truncating both domain/ip and user name when the they exceed their limits, instead of just domain/ip?

Reply:
Yes, I referred to the `fullnames_mode` flag.
There shouldn't really be magic numbers.